services/venice-controller/src/main/java/com/linkedin/venice/controller/VeniceParentHelixAdmin.java

```diff
@@ -2513,12 +2494,48 @@ public void updateStore(String clusterName, String storeName, UpdateStoreQueryPa
             ErrorType.BAD_REQUEST);
       }
 
-      final boolean writeComputeJustEnabled =
+      // When this config is true, it means partial update is enabled by request.
+      boolean partialUpdateJustEnabled =
           writeComputationEnabled.orElse(false) && !currStore.isWriteComputationEnabled();
-      if (writeComputeJustEnabled) {
+      if (partialUpdateJustEnabled) {
         // Dry-run generating Write Compute schemas before sending admin messages to enable Write Compute because Write
         // Compute schema generation may fail due to some reasons. If that happens, abort the store update process.
         addWriteComputeSchemaForStore(clusterName, storeName, true);
+      } else {
+        /**
+         * If a store: (1) Is being converted to hybrid (2) Is not partial update enabled for now. (3) Does not change
+         * partial update config in this request.
+         * It means partial update is not enabled, and there is no explict intention to change it. In this case, we will
+         * look up cluster config and based on the replication policy to enable partial update implicitly.
+         */
+        final boolean shouldEnablePartialUpdateBasedOnClusterConfig =
+            storeBeingConvertedToHybrid && (setStore.activeActiveReplicationEnabled
+                ? clusterConfig.isEnablePartialUpdateForHybridActiveActiveUserStores()
+                : clusterConfig.isEnablePartialUpdateForHybridNonActiveActiveUserStores());
+        if (!writeComputationEnabled.isPresent() && !currStore.isWriteComputationEnabled()
+            && shouldEnablePartialUpdateBasedOnClusterConfig) {
+          try {
+            addWriteComputeSchemaForStore(clusterName, storeName, true);
+            partialUpdateJustEnabled = true;
+            setStore.writeComputationEnabled = true;
+            updatedConfigSet.add(WRITE_COMPUTATION_ENABLED);
```

Comment:
What is the reason to change from `list` to `set` for `updatedConfig`?

Reply:
Eh. Because in my updated logic, some feature might get enabled twice (I think it is chunking, which got enabled by my new logic again maybe, and the list should contain two same params.

Reply:
I can try see if we actually don't need this change...after I make the above change to address the comment..